utils/src/main/java/se/fortnox/reactivewizard/util/LambdaCompiler.java

```diff
@@ -0,0 +1,77 @@
+package se.fortnox.reactivewizard.util;
+
+import java.lang.invoke.CallSite;
+import java.lang.invoke.LambdaMetafactory;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public class LambdaCompiler {
+    static boolean useLambdas = "true".equals(System.getProperty("useLambdas", "true"));
+
+    static <T> Supplier<T> compileLambdaSupplier(MethodHandles.Lookup lookup, MethodHandle methodHandle) throws Throwable {
+        if (!useLambdas) {
+            return () -> {
+                try {
+                    return (T) methodHandle.invoke();
+                } catch (Throwable e) {
```

Comment:
why catch on throwable? 
Error and its sub-classes are always unchecked exceptions, while the sub-classes of Exception can be either checked or unchecked. If the idea is to make checked exceptions unchecked and not to catch OOMs and other erros, then I guess you don't need throwable here.